### DISCUSSION TITLE
OSG 3.6.5 Use Light

### DISCUSSION THREAD

**Discussion by pusia1** [openscenegraph/OpenSceneGraph#1307] [Q&A]
I want to add a light source effect to the model in osg 3.6.5 version, but it doesn't seem to work with the default light and lightsource.
Does anyone know how?

**Comment by robertosfield** [comment]
Have a look at the osglight example:

https://github.com/openscenegraph/OpenSceneGraph/blob/master/examples/osglight/osglight.cpp

If you are new to the OSG then I recommend just skip learning the OSG and adopt it's successor the VSG.  The lighting support is both easier to use and more sophisticated - PBR, Cascaded shadow maps and seemly support for more than 8 lights that OpenGL/OSG fixed function pipeline support.

 https://github.com/vsg-dev/VulkanSceneGraph

**Reply by pusia1** [reply]
![image](https://github.com/openscenegraph/OpenSceneGraph/assets/63691030/11c7b755-df9a-4aae-b455-538f723518eb)


I don't think it's working

**Reply by robertosfield** [reply]
I am afraid I can't provide free fine grained support for OpenSceneGraph users these days, I'm just too busy with other more current work.  I will try to provide quick pointers for users, but they will need to figure stuff out from there.

Please realize the OpenSceneGraph is a legacy API, it's really not something your should be adopting for new projects so trying to get stuff working on API that you likely shouldn't be using anyway is a waste of your time. Please read #1291.
